File: C#/Classes/Factories/VehicleFactory.cs
using Factory.Enums;
using Factory.Interfaces;
using Factory.Vehicles;
using System;

namespace Factory.Factories
{
	public class VehicleFactory
	{
		Vehicle _vehicle;
		public void SetVehicleType(Vehicle vehicle)
		{
			_vehicle = vehicle;
		}

		public IVehicle GetVehicle()
		{
			switch (_vehicle)
			{
				case Vehicle.Car:
					return new Car();
				case Vehicle.SemiTruck:
					return new SemiTruck();
				default:
					throw new Exception("Unsupported Vehicle Specified");
			}
		}
	}
}

File: Python/VehicleFactory.py
from Vehicles import Car, SemiTruck
from Vehicle import Vehicle


class VehicleFactory:
    def __init__(self):
        self.vehicle = None
        pass

    def set_vehicle_type(self, vehicle):
        self.vehicle = vehicle

    def get_vehicle(self):
        if self.vehicle == Vehicle.CAR:
            return Car()
        elif self.vehicle == Vehicle.SEMI_TRUCK:
            return SemiTruck()
        else:
            raise Exception("Unsupported Vehicle Specified")
